index.js

```diff
@@ -48,6 +48,19 @@ module.exports = (function () {
 		return process.nextTick;
 	}
 
+	// Promise
+	if ((typeof Promise === "function" && typeof Promise.prototype.then === "function")) {
+		var ua = typeof window !== 'undefined' && window.navigator.userAgent.toLowerCase();
+		var isIOS = ua && /iphone|ipad|ipod|ios/.test(ua);
+		var promise = Promise.resolve();
+        var logError = console.error.bind(console);
+        var noop = function(){};
+		return function (cb) {
+			promise.then(cb).catch(logError)
+			if (isIOS) setTimeout(noop);
```

Comment:
> in problematic UIWebViews, Promise.then doesn't completely break, but
   it can get stuck in a weird state where callbacks are pushed into the
   microtask queue but the queue isn't being flushed, until the browser
   needs to do some other work, e.g. handle a timer. Therefore we can
   "force" the microtask queue to be flushed by adding an empty timer.